bundles/org.eclipse.equinox.simpleconfigurator/src/org/eclipse/equinox/internal/simpleconfigurator/ConfigApplier.java

```diff
@@ -91,7 +91,7 @@ void install(URL url, boolean exclusiveMode) throws IOException {
 		}
 
 		Set<Bundle> prevouslyResolved = getResolvedBundles();
-		Collection<Bundle> toRefresh = new ArrayList<>();
+		Collection<Bundle> toRefresh = new LinkedHashSet<>();
```

Comment:
Are duplicates possible?

Reply:
I don't think so, but this is passed to a method that performs "contains" checks and there the list was copied into a set (possibly to speedup the lookup) so I think it is better to build a set at the first place.